### PR TITLE
added log link for job controllers

### PR DESCRIPTION
**Pull request by cheld**
Added link to log view from jobs page.

I noticed that error handling is missing in one place. Not sure if it should be adapted to the new style.

### CODE REVIEW

**Comment by maciaszczykm** [comment]
> I noticed that error handling is missing in one place. Not sure if it should be adapted to the new style.

Where?

File: src/app/backend/resource/controller/controller.go [hunk 1]
```diff
@@ -266,16 +268,26 @@ func (self StatefulSetController) UID() types.UID {
...
+// Get controlled pods with owner reference
+func getControlledPods(k8sClient *client.Clientset, namespace string, uid types.UID) []v1.Pod {
 	allPods, _ := k8sClient.CoreV1().Pods(namespace).List(listEverything)
-	matchingPods := common.FilterPodsByOwnerReference(namespace, uid, allPods.Items)
+	return common.FilterPodsByOwnerReference(namespace, uid, allPods.Items)
+}
+
+// Get controlled pods with label
+func getControlledPodsForJob(k8sClient *client.Clientset, job batch.Job) []v1.Pod {
+	allPods, _ := k8sClient.CoreV1().Pods(job.Namespace).List(listEverything)
```
Comment (cheld) [inline src/app/backend/resource/controller/controller.go]: error dropped

**Comment by maciaszczykm** [comment]
It should not be dropped. It is better to return empty list than have null pointer dereference on `allPods.Items`.

**Comment by cheld** [comment]
Sure. I might have questions about the new error handling cocept.

**Comment by maciaszczykm** [comment]
@cheld Sure. It is not final implementation so any discussion is welcomed :)

**Comment by cheld** [comment]
PTAL

Sorry, I had to squash, becuase I messed up the history during rebase

File: src/app/backend/resource/controller/controller.go [hunk 2]
```diff
@@ -251,18 +267,38 @@ func (self StatefulSetController) UID() types.UID {
 }
 
 // GetLogSources is an implementation of the GetLogSources method from ResourceController interface.
-func (self StatefulSetController) GetLogSources(k8sClient *client.Clientset) LogSources {
+func (self StatefulSetController) GetLogSources(k8sClient *client.Clientset) (LogSources, error) {
```
Comment (floreks) [inline src/app/backend/resource/controller/controller.go]: I'm wondering why isn't this done same as for `Get` method. I mean that we don't need to pass client here. List of pods can already be retrieved in `getLogSourceFromController` method and passed here to do resource specific handling (filter by owner ref or filter pods for job).

Comment (cheld) [inline src/app/backend/resource/controller/controller.go]: Ok, makes sense. I will change

**Comment by cheld** [comment]
PTAL

File: i18n/messages-en.xtb [hunk 3]
```diff
@@ -362,7 +362,7 @@
   <translation id="4158586803314397594" key="MSG_ENDPOINT_CARDLIST_1" desc="(no description provided)">There are currently no Endpoints selected by this Service.</translation>
   <translation id="2771036553361600987" key="MSG_ENDPOINT_CARDLIST_2" desc="Label \'Source\' for the endpoint source column of the endpoints table (endpoints list page).">Host</translation>
   <translation id="7755291325340264720" key="MSG_ENDPOINT_CARDLIST_3" desc="Label \'Sub-object\' for the respective column of the endpoints table (endpoints list page).">Ports (Name, Port, Protocol)</translation>
-  <translation id="212373317303144699" key="MSG_ENDPOINT_CARDLIST_4" desc="Label \'Sub-object\' for the respective column of the endpoints table (endpoints list page).">Node name</translation>
+  <translation id="6378019040175268379" key="MSG_ENDPOINT_CARDLIST_4" desc="Label \'Sub-object\' for the respective column of the endpoints table (endpoints list page).">Node</translation>
```
Comment (cheld) [inline i18n/messages-en.xtb]: someone did not generate the translations before checkin?

Comment (maciaszczykm) [inline i18n/messages-en.xtb]: @kenan435 Did change his own label, but did not regenerate them. You should commit it.

File: src/app/backend/resource/controller/controller_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"reflect"
+	"testing"
+
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type Person struct {
```
Comment (floreks) [inline src/app/backend/resource/controller/controller_test.go]: Not used?

Comment (cheld) [inline src/app/backend/resource/controller/controller_test.go]: Oh sorry :) I tried something with go inheritence...

File: src/app/backend/resource/controller/controller.go [hunk 5]
```diff
@@ -220,11 +223,12 @@ func (self DaemonSetController) UID() types.UID {
 }
 
 // GetLogSources is an implementation of the GetLogSources method from ResourceController interface.
-func (self DaemonSetController) GetLogSources(k8sClient *client.Clientset) LogSources {
+func (self DaemonSetController) GetLogSources(allPods []v1.Pod) (LogSources, error) {
```
Comment (cheld) [inline src/app/backend/resource/controller/controller.go]: error shouldn't be needed any more

Comment (floreks) [inline src/app/backend/resource/controller/controller.go]: Remove them from interface method definition and they won't be.

**Comment by maciaszczykm** [comment]
What about link on list view? Can you add it? Check replica set list, because it is done here.

**Comment by cheld** [comment]
I always forget this button :)

**Pull request merged**